### ISSUE TITLE
Spurious "SameNameButDifferent" warning on Lombok code with `@Getter`

### ISSUE THREAD

**Issue by hakanai** [google/error-prone#3187]
For some Lombok code with a `@Getter` annotation, the "SameNameButDifferent" warning is triggered with an error message that seems to suggest it's treating the other types and annotations on the field as if they are field names. (?)

Related to #2982, #3094, but different because it's now a spurious warning instead of an exception.

```
...\src\main\java\com\example\Person.java:13: warning: [SameNameButDifferent] The name `@Getter` refers to [javax.annotation.Nullable, java.lang.SuppressWarnings, lombok.Generated, java.lang.String] within this file. It may be confusing to have the same name refer to multiple types. Consider qualifying them for clarity.
    @Getter
    ^
    (see https://errorprone.info/bugpattern/SameNameButDifferent)
```

Versions in play:
* Lombok 1.18.24
* ErrorProne 2.7.1, 2.13.1
* Java 11

[Repro](https://github.com/hakanai/errorprone-lombok-missingsummary) - **NOTE:** This repo was for reproducing a different issue, but it exhibits this issue also. You might want to comment out the MissingSummary line in `build.gradle.kts`, otherwise that causes an outright failure, but you should get the warning about SameNameButDifferent in the output regardless.

**Comment by hakanai** [comment]
What I see in the debugger before the warning is about to be emitted:

```
table = {HashBasedTable@13385} "{javax.annotation.Nullable={javax.annotation.Nullable=[com.sun.source.util.TreePath@59345113]}, lombok.Builder={lombok.Builder=[com.sun.source.util.TreePath@1e0e4bfc]}, lombok.Getter={lombok.Getter=[com.sun.source.util.TreePath@6102c3]}, Getter={lombok.Getter=[com.sun.source.util.TreePath@16fbb3f5, com.sun.source.util.TreePath@6d3cd4f0]}, @Getter={javax.annotation.Nullable=[com.sun.source.util.TreePath@1013300, com.sun.source.util.TreePath@97e4ef9, com.sun.source.util.TreePath@42512038, com.sun.source.util.TreePath@49c1f7d6], java.lang.SuppressWarnings=[com.sun.source.util.TreePath@41d044dd, com.sun.source.util.TreePath@3aa2c398], lombok.Generated=[com.sun.source.util.TreePath@71b97414, com.sun.source.util.TreePath@370042b2], java.lang.String=[com.sun.source.util.TreePath@74350e23]}, String={java.lang.String=[com.sun.source.util.TreePath@58a7bd3e]}, @Builder={java.lang.SuppressWarnings=[com.sun.source.util.TreePath@62047bea, com.sun.source.util.TreePath@57a4a0ca, com.sun.source.util.Tr"
 backingMap = {LinkedHashMap@13407}  size = 7
 factory = {HashBasedTable$Factory@13408} 
 columnKeySet = null
 rowMap = {StandardTable$RowMap@13409}  size = 7
  "javax.annotation.Nullable" -> {StandardTable$Row@13420}  size = 1
  "lombok.Builder" -> {StandardTable$Row@13422}  size = 1
  "lombok.Getter" -> {StandardTable$Row@13424}  size = 1
  "Getter" -> {StandardTable$Row@13426}  size = 1
  "@Getter" -> {StandardTable$Row@13428}  size = 4 👈
  "String" -> {StandardTable$Row@13430}  size = 1
  "@Builder" -> {StandardTable$Row@13432}  size = 6 👈
```

**Comment by hakanai** [comment]
At the first point during tree traversal, where one of the offending entries ends up in the table:

```
tree = {JCTree$JCIdent@13451} "Nullable"
treeSource = "@Getter"
symbol = {Symbol$ClassSymbol@13390} "javax.annotation.Nullable"
```

It almost seems like `treeSource` has just been determined incorrectly? And then `"@Getter"` ends up in the table where I'd expect `"Nullable"` instead.

**Comment by NZhuravlev** [comment]
Same here. You can also add `@Slf4j` to the list, which causes the same error.
